### ISSUE TITLE
[Select] Clearable variant is announced incorrectly

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3097]
## Current behavior 

Screen readers placed on [clearable select](https://storybook.sanomalearning.design/?path=/story/form-select--clearable) add name of "Clear selection" button to things that are announced on `sl-select-button`
It may be understood by user as label of select or option and can confuse them.

<img width="882" height="224" alt="Image" src="https://github.com/user-attachments/assets/5ac87d83-092c-49a9-a309-31e11088fc55" />

<img width="461" height="272" alt="Image" src="https://github.com/user-attachments/assets/667c7618-8217-4421-b647-5d1107cab565" />

Name of button is announced  as a part of combobox value because in DOM button is placed inside `sl-select-button`.

<img width="438" height="210" alt="Image" src="https://github.com/user-attachments/assets/649d1add-61fe-4268-9a36-a093d452836a" />

## Suggested solution 

The most reliable solution would be to place "Clear selection" button outside of `sl-select-button` in DOM similar to [Search field](https://storybook.sanomalearning.design/?path=/story/form-search-field--basic).

Other methods including:

1. change of `value` attribute with JavaScript 
2. adding `aria-valuetext` to override existing `value`
3. using `aria-activedescendant`

all failed on Chrome with VoiceOver, where screen reader still announced improper value of combobox.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
